crates/sui-keys/src/keystore.rs

```diff
@@ -129,12 +150,14 @@ impl<'de> Deserialize<'de> for FileBasedKeystore {
 
 impl AccountKeystore for FileBasedKeystore {
     fn sign_hashed(&self, address: &SuiAddress, msg: &[u8]) -> Result<Signature, signature::Error> {
-        Ok(Signature::new_hashed(
-            msg,
-            self.keys.get(address).ok_or_else(|| {
+        let (_, key) = self
+            .keys
+            .get(address)
+            .ok_or_else(|| {
                 signature::Error::from_source(format!("Cannot find key for address: [{address}]"))
-            })?,
-        ))
+            })?
+            .clone();
```

Comment:
You shouldn't need to clone if you select the right component of the tuple first.

```rust
Ok(Signature::new_hashed(
    msg,
    self.keys.get(address).ok_or_else(|| {
        signature::Error::from_source(format!("Cannot find key for address: [{address}]"))
    })?
    .1,
))
```